asv/util.py

```diff
@@ -294,9 +292,17 @@ def which(filename):
 
     candidates = []
     for location in locations:
+        for alternate_ext in ['.exe', '.bat']:
```

Comment:
This must be contained inside `if WIN` --- the stuff does not exist on other platforms.

Reply:
Also, this is simpler to rewrite so that you build the list of possible filenames in a separate `if WIN: ... else: ...` block above: 
```
if WIN:
    ...
    filenames = [filename + ext for ext in ('.exe', '.bat', '')]
else:
    ...
    filenames = [filename]
...
for location in locations:
    for fn in filenames:
        candidate = os.path.join(location, fn)
        ...